base/experiments/ad9910_ram.py

```diff
@@ -0,0 +1,206 @@
+from artiq.experiment import *
+from artiq.coredevice.ad9910 import *
+from jax import JaxExperiment
+import numpy as np
+from enum import Enum
+
+
+__all__ = ["AD9910RAM"]
+
+
+class RAMType(Enum):
+    """The type of data in the RAM.
+
+    Enums:
+        FREQ: Data are treated as frequencies (Hz).
+            The exact frequencies will be played-back.
+        PHASE: Data are treated as phases (turns)
+            Phase modulation will be performed.
+        AMP: Data are treated as amplitudes.
+            Amplitude modulation will be performed.
+        Polar: Data are treated as amplitudes & phase.
+            Polar modulation will be performed.
+    """
+    FREQ = 1
+    PHASE = 2
+    AMP = 3
+    POLAR = 4
+
+
+class RAMProfile:
+    RAM_SIZE = 1024
+
+    def __init__(self, dds, data, ramp_interval, ram_type, ramp_mode,
+                 base_frequency=0, base_phase=0, base_amplitude=0):
+        """Generate a RAM profile to a DDS channel.
+            Include optional parameters based on the "Data Source Priority" of
+            AD9910. See the switch case below to find out which optional
+            parameter is needed.
+
+        Args:
+            dds: The DDS that will playback the RAM profile.
```

Comment:
dds: AD9910 object, ....

Generally it is good to add the type of the argument.